mathml-converters/src/main/java/com/formulasearchengine/mathmltools/converters/POMConverter.java

```diff
@@ -0,0 +1,156 @@
+package com.formulasearchengine.mathmltools.converters;
+
+import com.formulasearchengine.mathmltools.converters.cas.POMLoader;
+import com.formulasearchengine.mathmltools.converters.cas.PomXmlWriter;
+import com.formulasearchengine.mathmltools.converters.exceptions.MathConverterException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.w3c.dom.Document;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.stream.XMLStreamException;
+import java.io.*;
+import java.lang.reflect.InvocationTargetException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.*;
+
+/**
+ * This class is a wrapper of the MLP (POM-project) class.
+ * It can convertToDoc mathematical expressions and creates
+ * a tagged convertToDoc tree. Furthermore, it uses the PomXmlWriter
+ * to convertToDoc it to XML trees.
+ */
+public class POMConverter implements IConverter {
+
+    private static final Logger LOG = LogManager.getLogger(POMConverter.class.getName());
+
+    private Path referenceDir;
+    private POMLoader pom;
+
+    public POMConverter() {
+    }
+
+    @Override
+    public void init() {
+        try {
+            pom = new POMLoader();
+            pom.init();
+        } catch (Exception e) {
+            LOG.error("Cannot instantiate POMLoader.", e);
+        }
+    }
+
+    private String parseLatexMathToStringXML(String latex) throws InvocationTargetException, IllegalAccessException, XMLStreamException, IOException {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+        LOG.debug("Parse latex expression by POM-Tagger.");
+        pom.parse(latex);
+
+        LOG.debug("Write XML to output stream");
+        PomXmlWriter.writeStraightXML(pom, outputStream);
+
+        LOG.debug("Convert output stream to string and close output stream.");
+        String out = outputStream.toString();
+        outputStream.close();
+        return out;
+    }
+
+    private synchronized Document parseLatexMathToDOM(String latex)
+            throws
+            IOException,
+            ParserConfigurationException,
+            InvocationTargetException,
+            IllegalAccessException,
+            XMLStreamException,
+            ExecutionException,
+            InterruptedException {
+        LOG.info("Parse latex string to document...");
+        // we using a trick and use the PomXmlWriter to directly
+        // write the output the DocumentBuilder input stream.
+
+        // Piped in and output streams (connect the output to the input stream)
+        LOG.trace("Init linked piped in and output streams");
+        PipedInputStream inputStream = new PipedInputStream();
```

Comment:
good practice to close auto-closeables with try().

Reply:
I agree.